### PR TITLE
New blog post: Wesh App: Share Contact and Send Message

### PR DESCRIPTION
**Pull request by iuricmp**
- adds a new blog post "Wesh App: Share Contact and Send Message".
- small code refactor on **pages/blog.tsx** to accommodate a large list of blogs.
- adds an enumerator on blog pages in order to expose the blog sequence to the user.

### CODE REVIEW

**Comment by LFGaming** [comment]
Hi @iuricmp,
for me at the bottom of the home page it now looks like this:
image
I'm not sure if this is because of some change you've made but in my clone of the main from yesterday I don't have this problem

**Comment by jefft0** [comment]
Same problem for me as for @LFGaming . "Messaging" is pushed off the left side of the frame.

File: pages/posts/share-contact-and-send-message.tsx [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+                  Let’s imagine that client2 has sent the contact request, so we call <code>receiveContactRequest</code> which we will define below. It returns <code>request</code>{" "}
+                  which has the account public key of client2, so we call <code>ContactRequestAccept</code>&nbsp; (
+                  <a href="https://buf.build/berty/weshnet/docs/main:weshnet.protocol.v1#weshnet.protocol.v1.ProtocolService.ContactRequestAccept">docs</a>) to accept the contact
+                  request and create the Contact group.
+                </p>
+                <p>
+                  Now we need to activate this group. We use the same account public key of client2 to identify the Contact group and call <code>GroupInfo</code>&nbsp; (
+                  <a href="https://buf.build/berty/weshnet/docs/main:weshnet.protocol.v1#weshnet.protocol.v1.ProtocolService.GroupInfo">docs</a>) to get the group’s public key, and
+                  then call
+                  <code>ActivateGroup</code>&nbsp;(
```
Comment (jefft0) [inline pages/posts/share-contact-and-send-message.tsx:167]: Please add a space between "call" and "ActivateGroup".

File: pages/posts/share-contact-and-send-message.tsx [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+}
+                    `}
+                  </code>
+                </pre>
+                <p>
+                  Similar to the previous function, this function takes the client1 which we created in <code>main</code> . It also takes the <code>groupInfo</code> object of the Contact group. (The
+                  &nbsp;<code>main</code> function already used this to activate the group.)
+                </p>
+                <p>
+                  As in the previous function, we want to subscribe to events. In the [previous blog post](https://wesh.network/posts/wesh-app-with-persistent-storage), we briefly
```
Comment (jefft0) [inline pages/posts/share-contact-and-send-message.tsx:283]: Need to use HTML `<a>` for the link of "previous blog post".

File: pages/posts/share-contact-and-send-message.tsx [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+                  <code className="language-go">
+                    {`
+	if len(os.Args) == 2 {
+		doClient2(os.Args[1])
+		return
+	}
+`}
+                  </code>
+                </pre>
+                <p>Now we can define the function doClient2 which is called if we run using the contact string. Paste the following function to the file contact.go and save it.</p>
```
Comment (jefft0) [inline pages/posts/share-contact-and-send-message.tsx:309]: Please change "doClient2" and "contact.go" to code font.

File: pages/posts/share-contact-and-send-message.tsx [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+                <pre>
+								<code className="language-go">
+                    {`
+	go mod tidy
+	go run .
+                    `}
+                  </code>
+                </pre>
+                <p>
+                  (You only need to do go mod tidy the first time.) It should print the contact string from client1, something like&nbsp;
```
Comment (jefft0) [inline pages/posts/share-contact-and-send-message.tsx:407]: Please change "go mod tidy" to code font.

File: pages/posts/share-contact-and-send-message.tsx [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+                <p>
+                  This function takes the client1 which we created in <code>main</code> and calls <code>ServiceGetConfiguration</code>&nbsp; (
+                  <a href="https://buf.build/berty/weshnet/docs/main:weshnet.protocol.v1#weshnet.protocol.v1.ProtocolService.ServiceGetConfiguration">docs</a>). Instead of getting
+                  the configuration’s peer ID as in previous examples, we get the public key of client1’s Account group. This public key is also in the shared contact, and is used
+                  by client2 to send a contact request to client1. To receive it, we call <code>GroupMetadataList</code>&nbsp; (
+                  <a href="https://buf.build/berty/weshnet/docs/main:weshnet.protocol.v1#weshnet.protocol.v1.ProtocolService.GroupMetadataList">docs</a>) with the Account group
+                  public key.
+                </p>
+                <p>
+                  Most API functions return a data structure, but a few like <code>GroupMetadataList</code> return a subscription stream like <code>subMetadata</code>. We use a
```
Comment (LFGaming) [inline pages/posts/share-contact-and-send-message.tsx:236]: In "We use a  `for` loop" there is a dubble space between a and for

**Comment by LFGaming** [comment]
On lines 236 and 237 of `share-contact-and-send-message.tsx` there is a dubble space in the site.

my suggestion seems to have dissapeared so i hope this is clear enough

File: pages/posts/share-contact-and-send-message.tsx [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+                  &nbsp;<code>for</code> loop and call
```
Comment (jefft0) [inline pages/posts/share-contact-and-send-message.tsx:237]: As mentioned by @LFGaming , there are two spaces before "for". Maybe remove the `&nbsp;` ?

File: pages/posts/share-contact-and-send-message.tsx [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+                  &nbsp;<code>subMetadata.Recv()</code> which blocks until it receives an event (or end of stream). As you build more complex apps, an event loop like this may handle
```
Comment (jefft0) [inline pages/posts/share-contact-and-send-message.tsx:238]: As mentioned by @LFGaming , there are two spaces before "subMetadata.Recv()". Maybe remove the `&nbsp;` ?

**Comment by LFGaming** [comment]
most places where there is a `&nbsp;` have dubble spaces

**Review by jefft0 — APPROVED** [review]

**Pull request merged**